async-nats/src/jetstream/context.rs

```diff
@@ -108,6 +109,16 @@ impl Context {
             .client
             .request(format!("{}.{}", self.prefix, subject), request)
             .await?;
+
+        let status = message.status.unwrap_or(StatusCode::OK);
+        if status.is_server_error() {
+            // TODO(caspervonb) return jetstream::Error with a status code when that type lands.
```

Comment:
Proposal: return `Response::Error` instead at this stage if status is any kind of error code (400-600)?

Reply:
Wondering if we should ignore all other `StatusCode`s. My intuition says it will lead to a debugging session.
I know that we did not handle them before, but still, WDYT?